consensus/pacemaker/module.go

```diff
@@ -243,6 +265,82 @@ func (m *pacemaker) NewHeight() {
 		)
 }
 
+// This is called each time the system wants a delayed signal to build a block
+func (m *pacemaker) RegisterMinBlockTimeDelay() {
+	// Discard any previous timer
+	// DISCUSS: This should not happen, Identify cases where an active timer has to be discarded, remove cancellation logic if none
+	if m.prepareStepDelayer.cancelFunc != nil {
+		m.prepareStepDelayer.cancelFunc()
+	}
+
+	m.prepareStepDelayer.blockProposed = false
+	m.prepareStepDelayer.deadlinePassed = false
+
+	minBlockTime := time.Duration(m.pacemakerCfg.MinBlockTimeMsec * uint64(time.Millisecond))
+	ctx, cancel := context.WithCancel(context.TODO())
+	m.prepareStepDelayer.cancelFunc = cancel
+
+	go func() {
+		select {
+		case <-ctx.Done():
+			return
+		case <-m.GetBus().GetRuntimeMgr().GetClock().After(minBlockTime):
+			m.prepareStepDelayer.m.Lock()
+			defer m.prepareStepDelayer.m.Unlock()
+
+			// After the timeout, if there was any candidate request waiting for a signal, tell it to build the block
```

Comment:
What is a `candidate request waiting for a signal`? Please update the comment for future readers.